onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc

```diff
@@ -149,25 +156,7 @@ bool QDQOpBuilder::IsOpSupportedImpl(const GraphViewer&,
     }
   }
 
-  return true;
-}
-
-bool QDQOpBuilder::HasSupportedInputsImpl(const GraphViewer&, const Node& node,
-                                          const emscripten::val& wnn_limits, const logging::Logger& logger) const {
-  const auto& input_defs = node.InputDefs();
-  const std::string_view op_type = node.OpType();
-  int32_t input0_type = 0;  // input data type
-  int32_t input1_type = 0;  // x_scale data type
-  int32_t input2_type = 0;  // x_zero_point data type
-  bool has_input2 = TensorExists(input_defs, 2);
-
-  if (!GetType(*input_defs[0], input0_type, logger) ||
-      !GetType(*input_defs[1], input1_type, logger) ||
-      (has_input2 && !GetType(*input_defs[2], input2_type, logger))) {
-    return false;
-  }
```

Comment:
Why remove this type validation?

Reply:
Oh, I saw it. I don't suggest to drop `IsOpSupportedImpl` and merge it to `HasSupportedInputsImpl` in this PR, many other ops have the same situation, we can do a central optimization in future.